staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test_utils.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
```

Comment:
Thank you for taking a look on to this.
 I can move [fieldmanager_test_utils.go](https://github.com/kubernetes/kubernetes/pull/110565/files/2c4bb9185fe51574f8ed4859095da686e0c8c7f5#diff-7c90fc050bd54038caa0435a3a7fd20c91f9fd1a9d624257783d47b1ebbba9ae) file to a package named `fieldmanagertest` but then I cannot use some of the [local structs and methods](https://github.com/Karthik-K-N/kubernetes/blob/1f986cc69d8ac5b579ccc2e6817fd0a07903692b/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/versionconverter.go#L38) of fieldmanager.

Also what name would you suggest for the new file?

Reply:
If we remove all this things, it's going to break apis for anyone using it, it's a little unfortunate.

Do you have an opinion @liggitt ? is it OK to move these files?

Reply:
I really don't think we should expose this for use outside the package unit tests

Reply:
> I really don't think we should expose this for use outside the package unit tests

I had completely misread what this code was doing 🤦 

If we need to make some of the "local structs and methods" available to the testing package, we should make them internal then? The filename could be `fieldmanagertest.go`? same as the package.

Reply:
Did you mean rename `fieldmanager_test_utils.go` to `fieldmanagertest.go` within the fieldmanager package?

Reply:
No, we can't expose within the `fieldmanager` package, we need to have a `fieldmanagertest` package. And you should move the things that are currently package private into an internal package so that you can use them from both package without exposing them publicly.